urle27f76cced_iconmaster5326_HPPPlus_tgz-pJ8-com_iconmaster_hppplus_HPPPlusJ8/src/com/iconmaster/hppplus/parse/element/block/ElementWhile.java

```diff
@@ -0,0 +1,127 @@
+
+package com.iconmaster.hppplus.parse.element.block;
+
+import com.iconmaster.hppplus.HPPPlus;
+import com.iconmaster.hppplus.Parseable;
+import com.iconmaster.hppplus.SourceRange;
+import com.iconmaster.hppplus.exception.HPPPlusException;
+import com.iconmaster.hppplus.exception.parse.DoExpectedException;
+import com.iconmaster.hppplus.exception.parse.EndExpectedException;
+import com.iconmaster.hppplus.exception.parse.SyntaxException;
+import com.iconmaster.hppplus.parse.Element;
+import com.iconmaster.hppplus.parse.Parser;
+import com.iconmaster.hppplus.tokenize.Token;
+import com.iconmaster.hppplus.tokenize.element.TokenWord;
+import java.util.ArrayList;
+
+/**
+ *
+ * @author iconmaster
+ */
+public class ElementWhile implements Element {
+    public class WhileData {
+        private Element cond;
+        private ArrayList<Element> block;
+        public WhileData(Element cond,ArrayList<Element> thenb) {
+            this.cond = cond;
+            this.block = block;
+        }
+    }
+        
+    private SourceRange range;
+    private Element cond;
+    private ArrayList<Element> block;
+    private int blockSize;
+    
+    public ElementWhile() {
+        
+    }
+    
+    public ElementWhile(Element cond,ArrayList<Element> block,int blockSize,SourceRange range) {
+        this.range = range;
+        this.cond = cond;
+        this.block = block;
+        this.blockSize = blockSize;
+    }
+
+    @Override
+    public boolean isMatch(ArrayList parseList, int at) {
+        if (at+3>=parseList.size()) {return false;}
+        if (parseList.get(at) instanceof TokenWord) {
+            Token e = (Token) parseList.get(at);
+            return "WHILE".equalsIgnoreCase((String) e.getParsedContent());
+        }
+        return false;
+    }
+
+    @Override
+    public Element match(ArrayList parseList, int at) throws HPPPlusException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.